lib/Pod/To/Text.rakumod

```diff
@@ -86,9 +86,11 @@ sub declarator2text($pod) {
     next unless $pod.WHEREFORE.WHY;
     my $what = do given $pod.WHEREFORE {
         when Method {
-            my @params=$_.signature.params[1..*];
-              @params.pop if @params.tail.name eq '%_';
-              'method ' ~ $_.name ~ signature2text(@params, $_.returns)
+            my @params = $_.signature.params[1..*];
+            with @params.tail.name {
+                @params.pop when '%_';
+            }
```

Comment:
This is probably relatively hot code.  This alternative is about 20x as fast:

    my @params = $_.signature.params.skip;
    @params.pop if @params.tail.name eq '%_';

Reply:
Thanks!

Reply:
Wait, nope. It is not equivalent.
The root of this issue is that `@params.tail.name` *can* be Nil and so it says "Compare with Nil is BAD", which is bogus. With this `with ...` construct we avoid working with Nil and your patch actually looks almost the same as it was, faster but having a bogus warning, no?

Reply:
is it not so that `@params.tail.name` can only be `Nil` if `@params == 0`?  if so, then:

    @params.pop if @params && @params.tail.name eq '%_';